website/src/components/MainLayout.jsx

```diff
@@ -45,21 +49,49 @@ const muiTheme = getMuiTheme({
  */
 class MainLayout extends Component {
   /**
+   * Constructor for the class
+   * @param {Object} props The props to be passed in.
+   * @returns {void}
+   */
+  constructor(props) {
+    super(props);
+  }
+
+  /**
+   * Checks whether user is logged in after component mounts.
+   * @returns {void}
+   */
+  componentWillMount() {
+    getAuthState(this.props.dispatch);
+  }
+
+  /**
    * @method MainLayout#render
    * @returns {HTML} Rendered layout
    */
   render() {
+    const nav = this.props.authState ? <DashNavBar /> : <LandingNavBar />;
+
     return (
       <MuiThemeProvider
        muiTheme={muiTheme}
       >
         <div>
-           <NavBar />
-           {this.props.children}
+           {nav}
+           {React.cloneElement(this.props.children, {authState: this.props.authState})}
```

Comment:
This line allows us to pass this.props.authState down from MainLayout --> Home without having to call getAuthState & mapping to props twice in both MainLayout & Home components.